### PR TITLE
feat(frontend/hooks): replace 3rd-party BroadcastChannel with native Web API equivalence

### PR DESCRIPTION
**Pull request by hainenber**
feat(frontend/hooks): replace 3rd-party BroadcastChannel usage with native Web API equivalence

### SUMMARY

Implementation of  2 years old TODO :D

Safari 15.4 was released 2 years ago ([source](https://developer.apple.com/documentation/safari-release-notes/safari-15_4-release-notes)) and so we can utilize its native Web API to replace a 3rd-party implementation.

Credit to Eric Wong's [SO post](https://stackoverflow.com/a/78491331) for the implementation. I'm just a copy-paster after all :D

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TESTING INSTRUCTIONS

All CI checks should pass

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/hooks/useTabId.ts [hunk 1]
```diff
@@ -18,17 +18,13 @@
...
-  type: 'REQUESTING_TAB_ID' | 'TAB_ID_DENIED';
-  tabId: string;
-}
-
-// TODO: We are using broadcast-channel to support Safari.
-// The native BroadcastChannel API will be supported in Safari version 15.4.
-// After that, we should remove this dependency and use the native API.
-const channel = new BroadcastChannel<TabIdChannelMessage>('tab_id_channel');
+const channel: StrictBroadcastChannel<TabIdChannelMessage> =
+  new BroadcastChannel('tab_id_channel');
```
Comment (justinpark) [inline superset-frontend/src/hooks/useTabId.ts:27]: nit: How about making the tab_id_channel value constant?

**Comment by rusackas** [comment]
Thanks for tackling this! This looks like a great improvement, and long overdue! 

Is there already (or should there be) a test making sure that messages are sent/received correctly?

**Comment by hainenber** [comment]
Unfortunately there's none but I can produce the expected result similar to what described in the original [PR](https://github.com/apache/superset/pull/18576) :D 

State key is kept between reloads for a dashboard/chart opened in Explore tab. Would be sufficient?


https://github.com/user-attachments/assets/f02233ef-99d1-41e0-8a7e-22e8d96ef9c5

**Review by rusackas — APPROVED** [review]
It might be nice to have added test coverage, but this is an improvement over the current state of affairs for the product, and moves away from package complexities toward native browser, support, so I call it a win :)

**Pull request merged**